### PR TITLE
Refactor HTTP request context

### PR DESCRIPTION
**Pull request by wxiaoguang**
# Background

I was working on some new enhancements, while I found the modules/context/context.go is quite messy, so I think it's better to refactor first before other works.

# Major Changes

* `HTMLString(name string, data interface{})` => `RenderToString(name base.TplName, data map[string]interface{})`, the first paramter is a `TplName`, and the old `HTMLString` naming conflicts with other function names like `HTML`/`PlainText`
* `TmplLoadTimes` is reworked. The old `TmplLoadTimes` in old `HTMLString` has no effects because the render doesn't use `ctx.Data`
* `NotFound(title string, err error)` and `ServerError(title string, err error)` related functions are refactored, because the `title` is not title, it only appears in log.
* use `PlainText(string)` and `PlainTextBytes([]byte)` to replace old `PlainText([]bytes)` and `HandleText(string)`
* remove unused `SignedUserName`
* use golang standard `type contextKeyType struct{}; var contextKey interface{} = contextKeyType{}` to replace `var contextKey interface{} = "default_context"`
* use `ctx.RespHeader()` to replace `ctx.Header()`, because `ctx.Header()` is not clear whether it is for request or response.

### CODE REVIEW

File: modules/context/context.go [hunk 1]
```diff
@@ -229,51 +225,47 @@ func (ctx *Context) RenderWithErr(msg string, tpl base.TplName, form interface{}
...
+func (ctx *Context) ServerError(logMsg string, logErr error) {
+	ctx.serverErrorInternal(logMsg, logErr)
 }
 
-func (ctx *Context) serverErrorInternal(title string, err error) {
-	if err != nil {
-		log.ErrorWithSkip(2, "%s: %v", title, err)
+func (ctx *Context) serverErrorInternal(logMsg string, logErr error) {
+	if logErr != nil {
+		log.ErrorWithSkip(2, "serverErrorInternal: %s: %v", logMsg, logErr)
```
Comment (KN4CK3R) [inline modules/context/context.go]: `serverErrorInternal: ` should be removed? This call does not log an error in `serverErrorInternal` but the provided error.

Comment (wxiaoguang) [reply modules/context/context.go]: Usually, I prefer to know where the error message comes from.

With the prefix:

```
2021/12/14 19:06:34 .../web/user/profile.go:35:GetUserByName() [E] notFoundInternal: GetUserByName: test error
```

Without the prefix:

```
2021/12/14 19:06:34 .../web/user/profile.go:35:GetUserByName() [E] GetUserByName: test error
```

For the first log, we can know that we get a 404 error at the first glance.

If you prefer not using the prefix, I will remove it.

Comment (KN4CK3R) [reply modules/context/context.go]: Thank you for removing it. I think it's just more noise in this case.

**Comment by wxiaoguang** [comment]
CI failure is not related. Other passed PRs also have such failure:

```
        testlogger.go:78: 2021/12/14 12:02:08 ...y/files/temp_repo.go:257:CommitTreeWithDate() [E] Unable to commit-tree in temporary repo: user2/initial-unsigned (integrations/gitea-integration-mysql/data/integrations/gitea-integration-mysql/tmp/local-repo/upload.git1503302879) Error: exit status 1
            	Stdout: 
            	Stderr: error: gpg failed to sign the data

        testlogger.go:78: 2021/12/14 12:02:08 .../api/v1/repo/file.go:387:handleCreateOrUpdateFileError() [E] UpdateFile: Unable to commit-tree in temporary repo: user2/initial-unsigned Error: exit status 1
            	Stdout: 
            	Stderr: error: gpg failed to sign the data
```

**Review by KN4CK3R — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**